django/contrib/admin/static/admin/js/admin/DateTimeShortcuts.js

```diff
@@ -205,10 +211,40 @@
 
             // Show the clock box
             clock_box.style.display = 'block';
+            
+            // Add focus loop for keyboard navigation
+            DateTimeShortcuts.addFocusLoop(clock_box);
+            
+            // Focus on the "Now" option for better screen reader experience
+            // Screen readers can then navigate the dialog naturally
+            const nowLink = clock_box.querySelector('ul.timelist li:first-child a');
+            if (nowLink) {
+                nowLink.focus();
+            } else {
+                // Fallback to first focusable element
+                const firstFocusableElement = clock_box.querySelector('a[href], button, [tabindex]:not([tabindex="-1"])');
+                if (firstFocusableElement) {
+                    firstFocusableElement.focus();
+                } else {
+                    clock_box.focus();
+                }
+            }
+            
             document.addEventListener('click', DateTimeShortcuts.dismissClockFunc[num]);
         },
         dismissClock: function(num) {
-            document.getElementById(DateTimeShortcuts.clockDivName + num).style.display = 'none';
+            const clock_box = document.getElementById(DateTimeShortcuts.clockDivName + num);
+            clock_box.style.display = 'none';
+            
+            // Remove focus loop
+            DateTimeShortcuts.removeFocusLoop(clock_box);
+            
+            // Restore focus to the trigger button
+            if (DateTimeShortcuts.lastFocusedElement) {
+                DateTimeShortcuts.lastFocusedElement.focus();
+                DateTimeShortcuts.lastFocusedElement = null;
+            }
```

Comment:
Rather than going custom for this, I think we should leverage `aria-controls`. Add the attribute to the time picker’s toggle button, then we can do:

```suggestion
            const clock_box = document.getElementById(DateTimeShortcuts.clockDivName + num);
            clock_box.style.display = 'none';
            const trigger = document.querySelector(`[aria-controls="${clock_box.id}"]`);
            trigger.focus();
```